.github/workflows/lint.yml

```diff
@@ -4,6 +4,8 @@ on:
     branches:
       - main
   pull_request:
+    branches:
```

Comment:
i think we want to run lint even if the base branch isn't main. (so that chained PRs still get linted)

Reply:
You're right, reverting this. I'm confused why golangci-lint passed in the previous PR but failed on main even though they're the same code. Not sure what to do here so just leaving it be.